deps/jerry/jerry-core/parser/js/js-lexer.c

```diff
@@ -1756,7 +1781,25 @@ lexer_construct_number_object (parser_context_t *context_p, /**< context */
   uint32_t literal_index = 0;
   prop_length_t length = context_p->token.lit_location.length;
 
-  if (context_p->token.extra_value != LEXER_NUMBER_OCTAL)
+  if(context_p->token.extra_value == LEXER_NUMBER_BINARY)
+  {
+    const uint8_t *src_p = context_p->token.lit_location.char_p;
+    const uint8_t *src_end_p = src_p + length - 1;
+
+    if(LEXER_TO_ASCII_LOWERCASE (src_p[1]) == LIT_CHAR_LOWERCASE_B)
+    {
+      src_p++;
+    }
+
+    num = 0;
+    do
+    {
+      src_p++;
+      num = num * 2 + (ecma_number_t) (*src_p - LIT_CHAR_0);
+    }
+    while (src_p < src_end_p);
```

Comment:
@ErosZy See https://github.com/Rokid/ShadowNode/blob/master/deps/jerry/jerry-core/parser/js/js-lexer.c#L1766-L1781, I think we should clean up the `if-else` cases.